src/Runner.Worker/ActionRunner.cs

```diff
@@ -159,7 +160,10 @@ public async Task RunAsync()
             if (ExecutionContext.Global.Container != null)
             {
                 // Make sure required container is already created.
-                ArgUtil.NotNullOrEmpty(ExecutionContext.Global.Container.ContainerId, nameof(ExecutionContext.Global.Container.ContainerId));
+                if (!FeatureFlagManager.IsHookFeatureEnabled())
```

Comment:
The feature flag just lets the user opt into the feature, I think we also want to check if they have set the env to use custom hooks as well